ui/app/AppLayouts/Chat/views/ChatMessagesView.qml

```diff
@@ -237,12 +252,15 @@ Item {
 
         model: messageStore.messagesModel
 
-        // Not Refactored Yet
-        //Component.onCompleted: scrollToBottom(true)
+        Component.onCompleted: chatLogView.scrollToBottom(true)
 
         delegate: MessageView {
             id: msgDelegate
 
+            // We wait to load all messages, because switching back to chats makes
+            // the scroll go crazy so it loads way too many messages making it slow
+            visible:  index < 10 || root.allMessagesLoaded
```

Comment:
@jrainville there are 2 properties which may help with delegates management: ListView::reuseItems and ListView::cacheBuffer. Did you play with these ? They may help and seems to be more elegant solution.